src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_InvalidExpression.cs

```diff
@@ -344,17 +344,20 @@ static void F()
 }
 ";
             string expectedOperationTree = @"
-IVariableDeclaratorOperation (Symbol: var x) (OperationKind.VariableDeclarator, Type: null, IsInvalid) (Syntax: 'x = () => F()')
-  Initializer: 
-    IVariableInitializerOperation (OperationKind.VariableInitializer, Type: null, IsInvalid) (Syntax: '= () => F()')
-      IAnonymousFunctionOperation (Symbol: lambda expression) (OperationKind.AnonymousFunction, Type: null, IsInvalid) (Syntax: '() => F()')
-        IBlockOperation (1 statements) (OperationKind.Block, Type: null, IsInvalid, IsImplicit) (Syntax: 'F()')
-          IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null, IsInvalid, IsImplicit) (Syntax: 'F()')
-            Expression: 
-              IInvocationOperation (void Program.F()) (OperationKind.Invocation, Type: System.Void, IsInvalid) (Syntax: 'F()')
-                Instance Receiver: 
-                  null
-                Arguments(0)
+    IVariableDeclaratorOperation (Symbol: var x) (OperationKind.VariableDeclarator, Type: null, IsInvalid) (Syntax: 'x = () => F()')
+      Initializer: 
+        IVariableInitializerOperation (OperationKind.VariableInitializer, Type: null, IsInvalid) (Syntax: '= () => F()')
+          IAnonymousFunctionOperation (Symbol: lambda expression) (OperationKind.AnonymousFunction, Type: null, IsInvalid) (Syntax: '() => F()')
+            IBlockOperation (1 statements) (OperationKind.Block, Type: null, IsInvalid, IsImplicit) (Syntax: 'F()')
+              IReturnOperation (OperationKind.Return, Type: null, IsInvalid, IsImplicit) (Syntax: 'F()')
```

Comment:
>IReturnOperation [](start = 14, length = 16)

Same comment.

Reply:
Every return expression in a lambda must be converted to the inferred return type of the lambda.  In this case we fail to infer a return type. Perhaps we should infer `void`?

---
In reply to: [328308503](https://github.com/dotnet/roslyn/pull/38750#discussion_r328308503) [](ancestors = 328308503)